core/src/main/java/com/linecorp/armeria/server/DependencyInjector.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+import com.linecorp.armeria.common.util.SafeCloseable;
+import com.linecorp.armeria.server.annotation.Decorator;
+import com.linecorp.armeria.server.annotation.DecoratorFactory;
+import com.linecorp.armeria.server.annotation.ExceptionHandler;
+import com.linecorp.armeria.server.annotation.RequestConverter;
+import com.linecorp.armeria.server.annotation.ResponseConverter;
+
+/**
+ * Injects dependencies that are specified in {@link RequestConverter#value()},
+ * {@link ResponseConverter#value()}, {@link ExceptionHandler#value()}, {@link Decorator#value()} and
+ * {@link DecoratorFactory#value()}.
+ * If the dependencies are not injected by this {@link DependencyInjector}, they are created via the default
+ * constructor, which does not have a parameter, of the classes.
+ */
+@UnstableApi
+public interface DependencyInjector extends SafeCloseable {
```

Comment:
Would it make sense to move this interface to `common` so we can use this for both client and server side? For example, we could introduce a client-side (or common) `@Decorator` annotation and let our users use it with Retrofit or our future Retrofit alternative.

Reply:
That's a good suggestion. 👍 Moved the package. 😉